bouncer/sentry.py

```diff
@@ -0,0 +1,31 @@
+"""Report exceptions to Sentry/Raven."""
+import raven
+
+
+def raven_tween_factory(handler, _):
+    """Return a tween that reports uncaught exceptions to Sentry."""
+    def raven_tween(request):
+        """Report uncaught exceptions to Sentry."""
+        try:
+            return handler(request)
+        except:
+            request.raven.captureException()
+            raise
+
+    return raven_tween
+
+
+def raven_client(request):
+    """Return the raven.Client object."""
+    request.add_finished_callback(
+        lambda request: request.raven.context.clear(),
+    )
+
+    return raven.Client(
+        dsn=request.registry.settings.get("sentry_dsn"),
+    )
+
+
+def includeme(config):
+    config.add_request_method(raven_client, name="raven", reify=True)
+    config.add_tween("bouncer.sentry.raven_tween_factory")
```

Comment:
Well, it's tricky. I'd leave it as is. The problem is that if your exception view is a catch-all that swallows exceptions, then it will break raven. But if it's catching a specific kind of exception which doesn't need reporting, then raven should be over it, not under it.